components/dashboard/src/settings/Teams.tsx

```diff
@@ -0,0 +1,767 @@
+                return { done: false };
+            }
+        }, opts);
+    }
+    const pollForAdditionalSlotsBought = (ts: TeamSubscription) => {
+        const opts: PollOptions<TeamSubscriptionSlotResolved[]> = {
+            backoffFactor: 1.2,
+            warningInSeconds: 40,
+            retryUntilSeconds: 120,
+            success: (result) => {
+                setSlots(result || []);
+                // showLoadingAnimation: false, // todo progress
+                // loadingTookLongerThanExpected: false // todo progress
+            },
+            stop: () => {
+                // this.closeModal() // todo progress
+
+            } 
+        };
+        poll<TeamSubscriptionSlotResolved[]>(1, async () => {
+            const freshSlots = await getGitpodService().server.tsGetSlots();
+            if (freshSlots.length > slots.length) {
+                return { done: true, result: freshSlots };
+            }
+            return { done: false };
+        }, opts);
+    }
+
+    const getPlan = (sub: TeamSubscription) => {
+        return Plans.getAvailableTeamPlans().filter(p => p.chargebeeId === sub.planId)[0];
+    }
+
+    const subscriptionMenu = (sub: TeamSubscription) => {
+        const result: ContextMenuEntry[] = [];
+        result.push({
+            title: 'Manage Members',
+            onClick: () => manageMemebers(sub)
+        })
+        result.push({
+            title: 'Add Members',
+            onClick: () => addMembers(sub)
+        })
+        result.push({
+            title: 'Invite Members',
+            onClick: () => inviteMembers(sub)
+        })
+        return result;
+    };
+
+    const showCreateTeamModal = () => {
+        const types = getAvailableSubTypes();
+        if (types && types.length > 0) {
+            setCreateTeamModal({ types })
+        }
+    }
+
+    const manageMemebers = (sub: TeamSubscription) => {
+        setManageTeamModal({ sub });
+    }
+
+    const addMembers = (sub: TeamSubscription) => {
+        setAddMembersModal({ sub });
+    }
+
+    const inviteMembers = (sub: TeamSubscription) => {
+        setInviteMembersModal({ sub });
+    }
+
+    const showBilling = () => {
+        if (chargebeeClient) {
+            chargebeeClient.openPortal();
+        }
+    }
+
+    return (<div>
+
+        <div className="flex flex-row">
+            <div className="flex-grow ">
+                <h3 className="self-center">All Teams</h3>
+                <h2>Manage teams and team members.</h2>
+            </div>
+            <div className="flex flex-end space-x-3">
+                {isChargebeeCustomer && (
+                    <button className="self-end my-auto" onClick={() => showBilling()}>Billing</button>
+                )}
+                {getActiveSubs().length > 0 && (
+                    <button className="self-end my-auto" disabled={!!pendingPurchase || getAvailableSubTypes().length === 0} onClick={() => showCreateTeamModal()}>Create Team</button>
```

Comment:
**thought**: We could skip disabling this button when a purchase is pending and throw an error within the modal if you already have a similar team with pending transaction. Alternatively, we could disable this when a user has one of a kind for all teams.

Any approach should be fine now. It should get better andmore clear once we [reimplement plans and pricing](https://www.notion.so/gitpod/Product-Roadmap-b9b5eac0a15147ac8d2dd25cf0519203#73ba4280a01446718fa2ad3ffe3d89da). Feel free to leave this as is for now. 🏀